spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -482,14 +489,14 @@ else if (result instanceof Message) {
 					iterableOfMessages = iterator.next() instanceof Message;
```

Comment:
the current approach assumes the `Iterable` is not `Iterable<?>` and treat the first element type as the indicator of the whole collection. but if we get a `List<Object>`, should we tell on a case-by-case basis instead?

Reply:
Let's leave it wit this assumption for now!
We can look into that later.
Let's concentrate on missed headers logic in this your PR!